modules/standard/IO.chpl

```diff
@@ -1699,11 +1753,18 @@ proc _modestring(mode:iomode) {
 }
 
 deprecated "open with a style argument is deprecated"
-proc open(path:string, mode:iomode, hints:iohints=IOHINT_NONE,
+proc open(path:string, mode:iomode, hints=ioHintSet.empty,
```

Comment:
This function is already deprecated, we don't need to provide an overload of it with the new type (and it didn't generate a deprecation warning for the `iohints` type anyways, right?)

Reply:
It looks like you do this a few times

Reply:
I think I created those versions so that users could be exposed to both deprecation warnings independently. So for example, if they happened to call the`open` overload with a style argument ***and*** the old iohints type (line 1763), they would first get a warning about iohints, and following the provided suggestion, they would switch to the version on line 1756. They would then compile again and get a second warning about the style argument, causing them to switch to the version on line 1784. 

In my mind, the alternative was for the warning on line 1762 to say something like "The 'iohints' type and style-argument are deprecated; please use a variant of 'open' that takes an 'ioHintSet' and no style argument instead." Which seemed too long.

Let me know if this changes your thinking at all. If not I'd certainly be happy to remove the extra overloads.

Reply:
I think it'd probably be frustrating to have to update twice, though.  We've previously gotten complaints about compilation errors hitting in waves instead of all at once, and there's not much we can do about that.

Reply:
Ok that makes sense. I'll remove those overloads then.